docs/src/main/sphinx/release/release-405.md

```diff
@@ -0,0 +1,148 @@
+  followed by partitioned and ordered `WINDOW` operators. {(issue)`15203`)
+
+## Security
+
+* Add support for string replacement as part of
+  [impersonation rules](system-file-auth-impersonation-rules). ({issue}`14962`)
+* Fix some `system.metadata` tables improperly showing the names of catalogs
+  which the user cannot access. ({issue}`14000`)
+* Fix `USE` statement improperly disclosing the names of catalogs and schemas
+  which the user cannot access. ({issue}`14208`)
+
+## BigQuery connector
+
+* Fix queries incorrectly executing with the project ID specified in the
+  credentials instead of the project ID specified in the `bigquery.project-id`
+  catalog property. ({issue}`14083`)
+
+## ClickHouse connector
+
+* Add context comments to `SELECT` queries. ({issue}`14500`)
+
+## Delta Lake connector
+
+* Improve performance and reduce storage requirements when running the `vacuum`
+  procedure on S3-compatible storage. ({issue}`15072`)
+* Improve performance of reading Parquet files for `boolean`, `tinyint`,
+  `short`, `int`, `long`, `float`, `double`, `short decimal`, `UUID`, `time`,
+  `decimal`, `varchar`, and `char` data types. The
+  `parquet.optimized-reader.enabled` catalog configuration property can be set
+  to `false` to disable the optimized implementation. ({issue}`14423`, {issue}`14667`)
+* Improve query performance when the `nulls fraction` statistic is not available
+  for some columns. ({issue}`15132`)
+* Allow registering existing table files in the metastore with the new
+  [`register_table` procedure](delta-lake-register-table). ({issue}`13568`)
+* Deprecate creating a new table with existing table content. This can be
+  re-enabled using the `delta.legacy-create-table-with-existing-location.enabled`
+  configuration property or the
+  `legacy_create_table_with_existing_location_enabled` session property. ({issue}`13568`)
+
+## Druid connector
+
+* Add context comments to `SELECT` queries. ({issue}`14500`)
+
+## Google Sheets connector
+
+* Rename the `credentials-path` configuration property to
+  `gsheets.credentials-path`, `metadata-sheet-id` to
+  `gsheets.metadata-sheet-id`, `sheets-data-max-cache-size` to
+  `gsheets.max-data-cache-size`, and `sheets-data-expire-after-write` to
+  `gsheets.data-cache-ttl`. ({issue}`15042`)
+
+## Hive connector
+
+* Improve performance of reading Parquet files for `boolean`, `tinyint`,
+  `short`, `int`, `long`, `float`, `double`, `short decimal`, `UUID`, `time`,
+  `decimal`, `varchar`, and `char` data types. The
+  `parquet.optimized-reader.enabled` catalog configuration property can be set
+  to `false` to disable the optimized implementation. ({issue}`14423`, {issue}`14667`)
+* Improve performance for queries which write data into multiple existing
+  partitions. ({issue}`15066`)
+* Disallow creating ACID tables when not using the Hive metastore. ({issue}`14673`)
+* Store query ID when creating a new schema. ({issue}`15174`)
```

Comment:
This seems like an implementation detail. What's the user-visible impact of this?

Reply:
User-visible impact is primarily the note below it, I'll remove.

Reply:
There's no user-visible impact. Removing this line is fine.